### PR TITLE
I CAN now see

### PR DESCRIPTION
**Pull request by ActuallyTaylor**
Added the ability for the HUDL to display data received from CAN on the display. Also updated the EVT-Core submodule so it includes the newest LCD driver.

### CODE REVIEW

File: src/HUDL/HUDL.cpp [hunk 1]
```diff
@@ -19,43 +20,94 @@ namespace log = EVT::core::log;
...
+        "RPM",
+        "Temp 1",
+        "Temp 2",
+        "Temp 3",
+        "Status 1",
+        "Pre Stat",
+        "Torque",
+    };
+
+    setDefaultSections(titles);
```
Comment (mjmagee991) [inline src/HUDL/HUDL.cpp]: This is setting the section string pointers to values in the array that is above, but this array only exists in the scope of this function, so it can get overwritten with other data. You should probably make `titles` a static member of the HUDL class.

File: src/HUDL/HUDL.cpp [hunk 2]
```diff
@@ -19,43 +20,94 @@ namespace log = EVT::core::log;
...
+    setTextForSection(4, tempTwo);
+    setTextForSection(5, tempThree);
+    //    setTextForSection(6, "ON");
+    //    setTextForSection(7, "Ready");
+    //    setTextForSection(8, "100 NM");
 }
 
-void HUDL::commWrite(uint8_t data) {
-    lcd.commandWrite(data);
+void HUDL::setTextForSection(uint8_t section, const char* text) {
```
Comment (mjmagee991) [inline src/HUDL/HUDL.cpp]: Does the HUDL really need to expose all this functionality?  Can some of these functions be made private or deleted?

File: src/HUDL/HUDL.cpp [hunk 3]
```diff
@@ -19,43 +20,94 @@ namespace log = EVT::core::log;
...
+    lcd.displayBitMapInArea(bitMap, bitMapWidth, bitMapHeight, page, column);
 }
 
-uint16_t HUDL::getObjectDictionarySize() const {
-    return OBJECT_DICTIONARY_SIZE;
+void HUDL::setEntireScreenBitMap(const uint8_t* bitMap) {
+    lcd.setEntireScreenBitMap(bitMap);
 }
 
-void HUDL::updateLCD() const {
```
Comment (mjmagee991) [inline src/HUDL/HUDL.cpp:50]: Why did you delete this function?

File: targets/HUDL/write_to_display.cpp [hunk 4]
```diff
@@ -63,7 +63,7 @@ int main() {
         // Clear LCD Screen
         // clear_lcd();
 
-        board.drivePixel(1, 1, 1, 255);
+        board.driveColumn(1, 1, 1, 255);
```
Comment (mjmagee991) [inline targets/HUDL/write_to_display.cpp]: Does this example still make sense given what `initLCD()` does? Is it worth keeping this example target here if there's already a bunch of examples in EVT-core?

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
Looks pretty good. I just have a couple thoughts on your implementation

File: src/HUDL/HUDL.cpp (hunk 3 above)
Comment (ActuallyTaylor) [reply src/HUDL/HUDL.cpp:50]: I didn't delete it, i just removed the const qualifier

File: include/HUDL/HUDL.hpp [hunk 5]
```diff
@@ -315,7 +282,40 @@ class HUDL {
             .Data = (uintptr_t) &totalVoltage,
         },
 
-        CO_OBJ_DIR_ENDMARK};
+        CO_OBJ_DIR_ENDMARK,
+    };
+
+    char* titles[9] = {
```
Comment (mjmagee991) [inline include/HUDL/HUDL.hpp]:
```suggestion
    static char* SECTION_TITLES[9] = {
```

File: src/HUDL/HUDL.cpp [hunk 6]
```diff
@@ -47,15 +35,42 @@ uint16_t HUDL::getObjectDictionarySize() const {
...
+    //    setTextForSection(2, "3000");
+    setTextForSection(3, tempOne);
+    setTextForSection(4, tempTwo);
+    setTextForSection(5, tempThree);
+    //    setTextForSection(6, "ON");
+    //    setTextForSection(7, "Ready");
+    //    setTextForSection(8, "100 NM");
+}
+
+void HUDL::setTextForSection(uint8_t section, const char* text) {
```
Comment (mjmagee991) [inline src/HUDL/HUDL.cpp]: If you aren't exposing these functions as public, there's not much reason to even have them private because they're a single line and don't hold any internal logic. Wherever you would use these functions, just make a direct call to the lcd function.

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
I have two more little things I'd like to see changed

**Comment by ActuallyTaylor** [comment]
[ef8e3ce](https://github.com/RIT-EVT/HUDL/pull/7/commits/ef8e3cee8734ad74ffe9a7d7c3ace74b01cb5a84) will not compile because it relies on [EVT-Core PR #74](https://github.com/RIT-EVT/EVT-core/pull/74).

**Review by mjmagee991 — APPROVED** [review]
Nice work

**Review by Ethangirard62 — COMMENTED** [review]
looks good

**Review by Ethangirard62 — APPROVED** [review]
